### ISSUE TITLE
[Burn-P3+ Bug]: Season is mandatory in some but not all datasheets

### ISSUE THREAD

**Issue by shreeramsenthi** [BurnP3/BurnP3Plus#50]
### What happened?

Season is now a mandatory to field for Daily Burning Hours sheet (with the default "All" season) to remove ambiguity, but is still optional in many (all?) other datasheets, which still leads to similar issues that are just harder to catch.

Maybe we should make it mandatory everywhere?

More specifically, this is is in all the advanced ignition datasheets as well as the spread event days and daily weather datasheets.

### What component are you seeing the problem on?

Syncrosim

### Approvals Process

- [ ] Testing For Issue
- [ ] Merge

**Comment by BadgerOnABike** [comment]
Seems reasonably as this would ensure the column always exists, even if its in an "all" state which is arguably either all or na, as it were.

**Comment by katieb1** [comment]
I've been working on fixing this, but it's not a super "small" fix. This will require a DB update (so a minor version change) and then will also affect how the input datasheets with the new mandatory column are filtered in all scripts and how the output datasheets are written. I'm thinking we wait until after the BP3+ course to implement? @LeonardoFrid

**Comment by shreeramsenthi** [comment]
makes sense to me! Also, bug was probably not the right label, just wanted to make sure this wasn't an accident.

**Comment by LeonardoFrid** [comment]
agreed on waiting till after the course for implementation.

**Comment by BadgerOnABike** [comment]
Do we want to do this prior to this next offering?

**Comment by katieb1** [comment]
@shreeramsenthi have you been working on updating this already? I thought I saw some commits related to this issue

**Comment by shreeramsenthi** [comment]
I made a couple fixes here: https://github.com/BurnP3/BurnP3Plus/commit/dcc857c6d3412ed6783605d68888750b26dc4957

to make sure season got filled into the Deterministic Ignition Location so the FGM's had a median Julian date for the fires, but this doesn't solve similar problems elsewhere.

Could certainly be made a priority if we want this done before the course.

**Comment by LeonardoFrid** [comment]
It would be good to get this fixed before the next offering. We have 2 months, so probably sufficient time? @shreeramsenthi and @katieb1, who is the best person to take the lead on this?

**Comment by shreeramsenthi** [comment]
I should have time, but feel free to go for it if you'd like @katieb1 .

One question for the folks at large though is how we'd like that implemented. Are we sure we want to make it required in the UI and bother folks to fill it explicitly in every compatible datasheet? I think this could also could break old libraries that don't have season specified in these tables unless we add some not super straightforward db updates for each of these tables.

I think an easier solution would be to add this to our error checking of inputs and just silently fill missing seasons with "All" during import and write them back to the result scenarios (the same way we handle resampling options). Also worth mentioning this still leaves the step of verifying that "All" is handed appropriately everywhere as a wildcard in all the sampling.

**Comment by katieb1** [comment]
@shreeramsenthi if you have time, that would be great! I'm a bit swamped at the moment...

I also like the idea of just silently filling it in if it doesn't exist. But yes, we'll have to do some extensive testing to make sure nothing is broken - another argument for having a comprehensive automated test suite!

**Comment by shreeramsenthi** [comment]
Can do, and sounds good!

**Comment by shreeramsenthi** [comment]
Hopefully fixed in: https://github.com/BurnP3/BurnP3Plus/pull/72
